File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, store: List[int], k: int) -> int:
        n = len(store)
        
        prefix_sums = [0] * (n + 1)
        for i in range(1, n + 1):
            prefix_sums[i] = prefix_sums[i - 1] + store[i - 1]
        
        suffix_sums = [0] * (n + 1)
        for i in range(1, n + 1):
            suffix_sums[i] = suffix_sums[i - 1] + store[n - i]
        
        max_score = 0
        
        for i in range(k + 1):
            max_score = max(max_score, prefix_sums[i] + suffix_sums[k - i])
        
        return max_score

File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.cpp
#include <iostream>
#include <numeric>
using namespace std;

class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int n = cardPoints.size();
        int lsum = accumulate(cardPoints.begin(), cardPoints.begin() + k, 0);
        int rsum = 0;
        int maximum = lsum;
        for(int i = 0; i < k; i++){
            lsum -= cardPoints[k - i - 1];
            rsum += cardPoints[n - i - 1];
            maximum = max(maximum, lsum + rsum);
        }
        return maximum;
    }
};